File: day1.py
calFile = open('day1.txt', 'r')

calString = calFile.read()

newList = calString.split('\n\n')

calList = []
for item in newList:
    calList += [item.strip().split('\n')]

for list in calList:
    for i in range(len(list)):
        list[i] = int(list[i])

maxList = []
for list in calList:
    maxList.append(sum(list))

maxCalories = max(maxList)

print(f'The reindeer who had the most calories had {maxCalories} calories')

orderList = sorted(maxList)[::-1]

topThree = orderList[0] + orderList[1] + orderList[2]

print(f'The sum of the top three calories is {topThree}')

File: day1.js
const fs = require('fs')
const calFile = fs.readFileSync('day1.txt').toString();

const rawCalData = calFile.split('\n\n');

var strCalData = [];
for (let i = 0; i < rawCalData.length; i++ ) {
    strCalData.push(rawCalData[i].split('\n'))
}

var calData = [];

for (let i = 0; i < strCalData.length; i++) {
    calData.push(strCalData[i].map(Number))
}

function sumList(numbers) {
    let result = 0;
    for (let i = 0; i < numbers.length; i++) {
      result += numbers[i];
    }
    return result;
  }
  
maxList = []
for (let i = 0; i < calData.length; i++) {
    maxList.push(sumList(calData[i]))
}


let maxCals = maxList.reduce((a,b) => {
    return Math.max(a, b);
});

console.log(`The reindeer with the most calories had ${maxCals} calories`)